File: Unidad_III/Brandon/ejemplo_2/C++/Auto/include/auto.cpp
#include <auto.hpp>

Auto::Auto( void )
{
    marca = "Mazda";
    modelo = "v3";
    year = 2019;
    kilometros = 10000;
    color = "Rojo";
};

Auto::Auto(std::string cMarca, std::string cModelo, int cYear, int cKilo, std::string cColor)
{
    marca = cMarca;
    modelo = cModelo;
    year = cYear;
    kilometros = cKilo;
    color = cColor;
}
void Auto::Pintar(std::string cColor)
{
    color = cColor;
    std::cout<<"El color del auto es: "<<Auto::color<<std::endl;
}
void Auto::Correr(int cKilo)
{
    kilometros = (cKilo+kilometros);
    std::cout << "El kilometraje del auto es: " << Auto::kilometros << std::endl;
}
void Auto::Modificar(std::string cModelo)
{
    modelo = cModelo;
    std::cout << "El modelo del auto es: " << Auto::modelo << std::endl;
}
void Auto::Checar(void)
{
    std::cout << "La marca del auto es: " << Auto::marca << std::endl;
    std::cout << "El modelo del auto es: " << Auto::modelo << std::endl;
    std::cout << "El año del auto es: " << Auto::year << std::endl;
    std::cout << "El kilometraje del auto es: " << Auto::kilometros << std::endl;
    std::cout << "El color del auto es: " << Auto::color << std::endl;
}

File: Unidad_III/Brandon/ejemplo_2/Python/Auto/include/auto.py
class Auto:
    def __init__(self,pMarca="Mazda",pModelo="v3",pYear=2019,pKilo=10000,pColor="Rojo"):
        self.marca=pMarca
        self.modelo=pModelo
        self.year=pYear
        self.kilo=pKilo
        self.color=pColor
    
    def Pintar(self,pColor):
        self.color=pColor
        print("El Nuevo color es: "+str(self.color))
    
    def Correr(self,pKilo):
        self.kilo=pKilo+self.kilo
        print("El auto viajo: "+str(self.kilo))
    
    def Modificar(self,pModelo):
        self.modelo=pModelo
        print("El Nuevo modelo es: "+str(self.modelo))
    
    def Checar(self):
        print("La marca es: "+str(self.marca))
        print("El modelo es: "+str(self.modelo))
        print("El año es: "+str(self.year))
        print("El kilometraje es: "+str(self.kilo))
        print("El color es: "+str(self.color))
